lib/helpers.bash

```diff
@@ -92,980 +96,934 @@ alias reload_completion="$(_make_reload_alias completion completion)"
-  _about 'reloads a profile file with error logs'
-  _group 'lib'
+function _bash-it-doctor-errors() {
+	_about 'reloads a profile file with error logs'
+	_group 'lib'
 
-  _bash-it-doctor $BASH_IT_LOG_LEVEL_ERROR
+	_bash-it-doctor "${BASH_IT_LOG_LEVEL_ERROR?}"
 }
 
-_bash-it-doctor-() {
-  _about 'default bash-it doctor behavior, behaves like bash-it doctor all'
-  _group 'lib'
+function _bash-it-doctor-() {
+	_about 'default bash-it doctor behavior, behaves like bash-it doctor all'
+	_group 'lib'
 
-  _bash-it-doctor-all
+	_bash-it-doctor-all
 }
 
-_bash-it-profile-save() {
-  _about 'saves the current configuration to the "profile" directory'
-  _group 'lib'
-
-  local name=$1
-  while [ -z "$1" ]; do
-    read -r -e -p "Please enter the name of the profile to save: " name
-    case $name in
-      "")
-        echo -e "\033[91mPlease choose a name.\033[m"
-        ;;
-      *)
-        break
-        ;;
-    esac
-  done
-
-  local profile_path="${BASH_IT}/profiles/${name}.bash_it"
-  if [ -f "$profile_path" ]; then
-    echo -e "\033[0;33mProfile \"$name\" already exists.\033[m"
-    while true; do
-      read -r -e -n 1 -p "Would you like to overwrite existing profile? [y/N] " RESP
-      case $RESP in
-        [yY])
-          echo -e "\033[0;32mOverwriting profile \"$name\"...\033[m"
-          rm "$profile_path"
-          break
-          ;;
-        [nN] | "")
-          echo -e "\033[91mAborting profile save...\033[m"
-          return 1
-          ;;
-        *)
-          echo -e "\033[91mPlease choose y or n.\033[m"
-          ;;
-      esac
-    done
-  fi
-
-  local something_exists
-  echo "# This file is auto generated by Bash-it. Do not edit manually!" > "$profile_path"
-  for subdirectory in "plugins" "completion" "aliases"; do
-    local component_exists=""
-    echo "Saving $subdirectory configuration..."
-    for f in "${BASH_IT}/$subdirectory/available/"*.bash; do
-      _bash-it-determine-component-status-from-path "$f"
-      if [ "$enabled" == "x" ]; then
-        if [ -z "$component_exists" ]; then
-            # This is the first component of this type, print the header
-            component_exists="yes"
-            something_exists="yes"
-            echo "" >> "$profile_path"
-            echo "# $subdirectory" >> "$profile_path"
-        fi
-        echo "$subdirectory $enabled_file_clean" >> "$profile_path"
-      fi
-    done
-  done
-  if [ -z "$something_exists" ]; then
-    echo "It seems like no configuration was enabled.."
-    echo "Make sure to double check that this is the wanted behavior."
-  fi
-
-  echo "All done!"
-  echo ""
-  echo "Profile location: $profile_path"
-  echo "Load the profile by invoking \"bash-it profile load $name\""
+function _bash-it-profile-save() {
+	_about 'saves the current configuration to the "profile" directory'
+	_group 'lib'
+
+	local name="${1:-}"
+	while [[ -z "$name" ]]; do
+		read -r -e -p "Please enter the name of the profile to save: " name
+		case $name in
```

Comment:
Till 627 there are a few unquoted variables (maybe a few elsewhere too, which a missed). I tend to be scrupulous about those 😄

Reply:
Yes! I am generally a "quote literally every single thing" kind of guy, but sometimes I get tired. 😉 Technically, variables in `case` statements don't need to be quoted but like I said I like to quote everything in Bash so I'll re-check for anything else I've left unquoted. (Zsh doesn't word split variables, so simply doesn't have this problem...)

We use `shellcheck` to look for this kind of thing, but it won't flag on `case` because it knows it's ok, so I didn't get my attention drawn to it.

Reply:
Ok, I've got these quoted on my branch but I haven't pushed it here yet. Maybe later today 😃

Reply:
I wasn't aware about the quoting in `case` statements, thank you for this information!